Turtle/turtle.js

```diff
@@ -0,0 +1,131 @@
+class Turtle {
+  constructor(x, y) {
+    this.x = x;
+    this.y = y;
+    this.history = [[x, y]]; // records all the movements.
+    this.direction = "east"; // this is starting position.
+  }
+
+  forward(steps) {
+    let start = this.history[this.history.length - 1];
+
+    switch (this.direction) {
+      case "east":
+        for (let i = 0; i < steps; i += 1)
+          this.history.push([start[0] + i, start[1]]);
+        break;
+
+      case "west":
```

Comment:
all these other hard coded strings: "west", "south", "north" should also be placed in constants